pkg/store/consul/dsstore/consul_store.go

```diff
@@ -221,28 +242,96 @@ func (s *ConsulStore) MutateDS(
 	if err != nil {
 		return fields.DaemonSet{}, consulutil.NewKVError("cas", dsPath, err)
 	}
+
 	if !success {
 		return fields.DaemonSet{}, CASError(dsPath)
 	}
+
+	return ds, nil
+}
+
+// MutateDSTxn adds a check-and-set operation to the passed transaction to
+// perform the mutation requested via the mutator function.
+// TODO: replace all calls of MutateDS with this
+func (s *ConsulStore) MutateDSTxn(
+	ctx context.Context,
+	id fields.ID,
+	mutator func(fields.DaemonSet) (fields.DaemonSet, error),
+) (fields.DaemonSet, error) {
+	ds, metadata, err := s.Get(id)
+	if err != nil {
+		if err == NoDaemonSet {
+			// we need to pass this through so callers can distinguish this from other errors
+			return fields.DaemonSet{}, err
+		}
+
+		return fields.DaemonSet{}, util.Errorf("Error getting daemon set: %v", err)
+	}
+
+	ds, err = mutator(ds)
+	if err != nil {
+		return fields.DaemonSet{}, util.Errorf("Error mutating daemon set: %v", err)
+	}
+	if ds.ID != id {
+		// If the user wants a new uuid, they should delete it and create it
+		return fields.DaemonSet{},
+			util.Errorf("Explicitly changing daemon set ID is not permitted: Wanted '%s' got '%s'", id, ds.ID)
+	}
+	if err := checkManifestPodID(ds.PodID, ds.Manifest); err != nil {
+		return fields.DaemonSet{}, util.Errorf("Error verifying manifest pod id: %v", err)
+	}
+
+	rawDS, err := json.Marshal(ds)
+	if err != nil {
+		return fields.DaemonSet{}, util.Errorf("Could not marshal DS as json: %s", err)
+	}
+
+	dsPath, err := s.dsPath(id)
+	if err != nil {
+		return fields.DaemonSet{}, util.Errorf("Error getting daemon set path: %v", err)
+	}
+
+	err = transaction.Add(ctx, api.KVTxnOp{
+		Verb:  api.KVCAS,
+		Key:   dsPath,
+		Index: metadata.LastIndex,
+		Value: rawDS,
```

Comment:
that's the intention. This change is just making all the ops use transactions so we can throw an audit log record into the transaction later